src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -55,14 +59,18 @@ public function load($data, $type = null)
 
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
             $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-            $normalizedShortName = Inflector::pluralize(Inflector::tableize($resourceMetadata->getShortName()));
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            if (null === $resourceShortName) {
+                throw new InvalidResourceException(sprintf('Resource "%s" has no short name defined.', $resourceClass));
```

Comment:
Why should we not expect it? Because of the single responsibility principle? Still, lots of things won't work without a shortname, and I thought it was mandatory, or at least required in the ResourceMetadata for it to be valid.